pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -425,3 +400,91 @@ func proxyMapToEnvVars(proxyConfig map[string]string) []corev1.EnvVar {
 	sort.Slice(envVars, func(i, j int) bool { return envVars[i].Name < envVars[j].Name })
 	return envVars
 }
+
+func gracefulTerminationDuration(infrastructureLister configlisterv1.InfrastructureLister) (watchTerminationDuration int) {
+	// 135s is our default value
+	//   the initial 70s is reserved fo the minimal termination period
+	//   additional 60s for finishing all in-flight requests
+	//   an extra 5s to make sure the potential SIGTERM will be sent after the server terminates itself
+	watchTerminationDuration = 135
+
+	infra, err := infrastructureLister.Get("cluster")
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+		}
+		// return defaults if there is no infrastructure resource or we encountered an error
+		return watchTerminationDuration
+	}
+	switch infra.Spec.PlatformSpec.Type {
+	case configv1.AWSPlatformType:
+		// AWS has a known issue: https://bugzilla.redhat.com/show_bug.cgi?id=1943804
+		// We need to extend the time needed for an LB to notice and remove unhealthy instance.
+		// Once the mentioned issue is resolved this code must be removed and the default values applied.
+		//
+		// The initial 210s is reserved for the minimal termination period.
+		// Additional 60s for finishing all in-flight requests.
+		// An extra 5s to make sure a potential SIGTERM will be sent after a server terminates itself.
+		watchTerminationDuration = 275
+	}
+	return watchTerminationDuration
+}
+
+func platformSpecificConfigOverrides(infrastructureLister configlisterv1.InfrastructureLister) []byte {
+	infra, err := infrastructureLister.Get("cluster")
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+		}
+		return nil
+	}
+	switch infra.Spec.PlatformSpec.Type {
+	case configv1.AWSPlatformType:
+		// AWS has a known issue: https://bugzilla.redhat.com/show_bug.cgi?id=1943804
+		// We need to extend the shutdown-delay-duration so that an NLB has a chance to notice and remove unhealthy instance.
+		// Once the mentioned issue is resolved this code must be removed and default values applied
+		return v410_00_assets.MustAsset("v4.1.0/config/aws-overrides.yaml")
+	}
+	return nil
+}
+
+type kasTemplate struct {
+	Image                       string
+	OperatorImage               string
+	Verbosity                   string
+	GracefulTerminationDuration int
+}
+
+func manageTemplate(rawTemplate string, imagePullSpec string, operatorImagePullSpec string, infrastructureLister configlisterv1.InfrastructureLister, operatorSpec *operatorv1.StaticPodOperatorSpec) (string, error) {
```

Comment:
podManifestFromTemplate

Reply:
this function is not tied to pods manifest it can manage any manifest. Seems `manageTemplate` is more correct.

Reply:
the word "manage" is wrong. Call it `completeTemplate` or `fromTemplate`. Manage is something with side-effects.